desktop_version/src/KeyPoll.cpp

```diff
@@ -496,3 +496,20 @@ bool KeyPoll::controllerWantsDown(void)
 {
     return buttonmap[SDL_CONTROLLER_BUTTON_DPAD_DOWN] || yVel > 0;
 }
+
+int KeyPoll::controllerRumble(Uint16 intensity, Uint32 duration_ms)
+{
+    if (game.rumble)
+    {
+        SDL_Event evt;
+        SDL_GameController *toOpen = SDL_GameControllerOpen(evt.cdevice.which);
```

Comment:
You open a controller handle, but don't close it later. You probably shouldn't open a handle every single time rumble is called, as well.

Reply:
This _slightly_ works because the handle, based on the instance ID, should already exist, and SDL_Quit clears out all refcounts from what I remember.

That said: This `evt` value isn't initialized anyway, so instead what we can do is loop through the existing controllers opened via SDL_GameControllerOpen in the hotplugging routines, avoiding the refcount and the uninitialized evt.